Added Win Condition for Finite Levels

maiuryn opened the pull request:
Created a prefab that will load a level select scene when touched.

maiuryn committed:
Added Win Condition for Finite Levels

adenprince commented on Assets/Scripts/Game Controllers/WinConditionController.cs:
```diff
@@ -0,0 +1,17 @@
+        if(collision.name == "Player")
+        {
+            // Loads pause scene
+            SceneManager.LoadScene(0);
```
This should use the scene name as a string instead of the scene build index so that it's clear which scene is being loaded. Then, the comment can be removed.

adenprince replied:
- [x] Update

adenprince commented on Assets/Scripts/Game Controllers/WinConditionController.cs:
```diff
@@ -0,0 +1,17 @@
+{
+    private void OnTriggerEnter2D(Collider2D collision)
+    {   
+        Debug.Log(collision.name);
```
This log can be removed.

adenprince commented on Assets/Scripts/Game Controllers/WinConditionController.cs:
```diff
@@ -0,0 +1,17 @@
+using System.Collections;
+using System.Collections.Generic;
```
The first two using directives are unused and can be removed.

adenprince commented on Assets/Scripts/Game Controllers/WinConditionController.cs:
```diff
@@ -0,0 +1,17 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class WinConditionController : MonoBehaviour
```
I think `WinConditionController.cs` should be renamed to `LevelEndTriggerController.cs`, and I also think that `Win Condition.prefab` should be renamed to `Level End Trigger.prefab`. The current names are fine with me though.

adenprince commented on Assets/Scripts/Game Controllers/WinConditionController.cs:
```diff
@@ -0,0 +1,17 @@
+    private void OnTriggerEnter2D(Collider2D collision)
+    {   
+        Debug.Log(collision.name);
+        if(collision.name == "Player")
```
It's probably better practice to use a tag or a layer here so that we don't rely on the player GameObject having a certain name ([Unity documentation on tags and layers](https://docs.unity3d.com/Manual/class-TagManager.html)). This can be left as is. I just wanted to point it out.

adenprince commented on Assets/Scenes/LevelSelectScene.unity:
```diff
@@ -0,0 +1,209 @@
+%YAML 1.1
```
I think this scene should be renamed to `LevelSelect.unity` so that the pause menu level select button works. Alternatively, the scene name used for that button could be changed to "LevelSelectScene".

adenprince reviewed (changes requested):
Looks good overall. I made a task for changing the `LoadScene` method call in `WinConditionController.cs` and some small optional suggestions. Once that task is complete, this pull request should be good to merge.
After this pull request merges, I think we will need to leave issue #6 open until the level completion message is added.

maiuryn committed:
Update Win Condition

Updated trigger to happen on collision with player tag, Changed scene change to target scene name.

adenprince commented on Assets/Scripts/Game Controllers/LevelEndTriggerController.cs:
```diff
@@ -0,0 +1,13 @@
+{
+    private void OnTriggerEnter2D(Collider2D collision)
+    {   
+        if(collision.tag == "Player")
```
It's recommended to use the `CompareTag` method instead of `==` for tag comparison ([documentation](https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/main/doc/UNT0002.md)).

adenprince commented on Assets/Prefabs/Level End Trigger.prefab:
```diff
@@ -0,0 +1,126 @@
+%YAML 1.1
```
I think the Sprite Renderer component should be removed from this prefab. The bounds are already visible in the Editor because of the Box Collider 2D component.

adenprince commented on Assets/Scenes/SampleScene.unity:
```diff
@@ -412,6 +412,10 @@ PrefabInstance:
       objectReference: {fileID: 0}
+    - target: {fileID: 5215622920849717728, guid: c9cb149ef1a6c1d49b10c17aad617566, type: 3}
+      propertyPath: m_TagString
+      value: Player
```
The Player tag change needs to be applied to the Player prefab. Currently, this tag change is only on the Player prefab instance in `SampleScene.unity`. See [this page](https://docs.unity3d.com/Manual/EditingPrefabViaInstance.html) for information about applying an override to a prefab.

adenprince commented on Assets/Scenes/SampleScene.unity:
```diff
@@ -631,6 +635,79 @@ RectTransform:
+    m_Modifications:
+    - target: {fileID: 2617945921448679910, guid: 309730eacf554f04e991349c77ea6ee5, type: 3}
+      propertyPath: m_Name
+      value: Win Condition
```
The Level End Trigger prefab instance in `SampleScene.unity` should be renamed from Win Condition to Level End Trigger.

adenprince reviewed (changes requested):
I wrote comments about changes I think should be made before merging.

maiuryn committed:
Merge branch 'premain' into 6_FiniteWinCondition

maiuryn committed:
Merge branch 'premain' into 6_FiniteWinCondition

maiuryn committed:
Update Level End Trigger

adenprince reviewed (approved):
Looks good, thank you.
